### PR TITLE
feat: mongo error rework

### PR DESCRIPTION
**Pull request by pichoemr**
Closes #

## Checklist
- [x] I didn't over-scope my PR
- [x] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I did not include breaking changes
- [ ] I made my own code-review before requesting one

### I included unit tests that cover my changes
  - [x] 👍 yes
  - [ ] 🙅 no, because they aren't needed
  - [ ] 🙋 no, because I need help

### I added/updated the documentation about my changes

- [ ] 📜 README.md
- [ ] 📕 docs/*.md
- [ ] 📓 docs.lenra.io
- [x] 🙅 no documentation needed

### CODE REVIEW

File: lib/mongo_storage.ex [hunk 1]
```diff
@@ -290,7 +290,7 @@ defmodule ApplicationRunner.MongoStorage do
       [_, hex_id] ->
         case BSON.ObjectId.decode(hex_id) do
           :error ->
-            BusinessError.not_an_object_id_tuple()
+            {:error, BusinessError.not_an_object_id_tuple()}
```
Comment (jonas-martinez) [inline lib/mongo_storage.ex]:
```suggestion
            BusinessError.not_an_object_id_tuple()
```

Why do you add `:error` ? It is already present in the call to `BusinessError` thanks to `_tuple()`.

This looks clean like this but if you expand the call to BusinessError to its real value it looks odd.
{:error, {:error, ...}}

File: lib/mongo_storage.ex [hunk 2]
```diff
@@ -347,12 +347,13 @@ defmodule ApplicationRunner.MongoStorage do
       {:ok, session_uuid}
     else
       :no ->
-        BusinessError.error_during_transaction_start_tuple(%{
-          error: "uuid already used, please try again"
-        })
+        {:error,
```
Comment (jonas-martinez) [inline lib/mongo_storage.ex]: Same thing here and at line 351

**Review by jonas-martinez — CHANGES_REQUESTED** [review]

File: lib/mongo_storage.ex [hunk 3]
```diff
@@ -347,12 +347,13 @@ defmodule ApplicationRunner.MongoStorage do
...
+         BusinessError.error_during_transaction_start_tuple(%{
+           error: "uuid already used, please try again"
+         })}
 
       {:error, msg} ->
-        BusinessError.error_during_transaction_start_tuple(%{error_message: msg})
+        {:error, BusinessError.error_during_transaction_start_tuple(%{error_message: msg})}
```
Comment (pichoemr) [inline lib/mongo_storage.ex]:
```suggestion
        BusinessError.error_during_transaction_start_tuple(%{error_message: msg})
```

File: lib/mongo_storage.ex [hunk 4]
```diff
@@ -347,12 +347,13 @@ defmodule ApplicationRunner.MongoStorage do
       {:ok, session_uuid}
     else
       :no ->
-        BusinessError.error_during_transaction_start_tuple(%{
-          error: "uuid already used, please try again"
-        })
+        {:error,
+         BusinessError.error_during_transaction_start_tuple(%{
+           error: "uuid already used, please try again"
+         })}
```
Comment (pichoemr) [inline lib/mongo_storage.ex]:
```suggestion
         BusinessError.error_during_transaction_start_tuple(%{
           error: "uuid already used, please try again"
         })
```

**Review by jonas-martinez — DISMISSED** [review]
LGTM

**Review by taorepoara — APPROVED** [review]

**Pull request merged**